### PR TITLE
Feature/aws cf s3 templates

### PR DESCRIPTION
**Pull request by pixelfields**

### CODE REVIEW

File: aws_cf_templates/cz-pycon-2015-hosted-zone.template [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            "Value": { "Ref": "czPycon2015HostedZone" },
+            "Description": "Hosted zone ID"
+        },
+        "DomainName": {
+            "Value": { "Ref": "czPycon2015DNSRecord" },
+            "Description": "The value of the domain name of the record set"
+        }
+    }
+
+}
```
Comment (tricoder42) [inline aws_cf_templates/cz-pycon-2015-hosted-zone.template]: Please add empty new line and the end of all files.

File: aws_cf_templates/cz-pycon-2015-hosted-zone.template [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            "Description": "The type of record to add",
+            "Default": "A",
+            "AllowedValues": [
+                "A"
+            ]
+        },
+        "S3BucketHostedZoneId": {
+            "Type": "String",
+            "Description": "The hosted zone ID. For Amazon S3, use the hosted zone ID for your bucket's website endpoint",
+            "Default": "Z1BKCTXD74EZPE"
```
Comment (tricoder42) [inline aws_cf_templates/cz-pycon-2015-hosted-zone.template:22]: This ID is the same for all S3 buckets on AWS, right?

**Comment by tricoder42** [comment]
Looks good, thank you very much!

Could you please provide some example, how do you use this templates?

File: aws_cf_templates/cz-pycon-2015-hosted-zone.template (hunk 1 above)
Comment (pixelfields) [inline aws_cf_templates/cz-pycon-2015-hosted-zone.template]: Done at bd0af84621b71f33378fb9fd8f9bb52e9b0d0f36

File: aws_cf_templates/cz-pycon-2015-hosted-zone.template (hunk 2 above)
Comment (pixelfields) [inline aws_cf_templates/cz-pycon-2015-hosted-zone.template:22]: This value is valid for all S3 buckets in the "eu-west-1" region. If you will deploy the template in another region, you must change this value to reflect region S3 website endpoint. Please see http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region . For example if you will deploy a stack in the "us-east-1" region, right value is Z3AQBSTGFYJSTF .

**Comment by pixelfields** [comment]
@tricoder42 For learn how to deploy a CF template and create a stack by AWS Cloud Formation Console please follow this user guide: http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-console-create-stack.html

CF templates are parameterized, so you can use them for different stacks, but default values for params reflect your requirements:
- DomainName param has default value "pycon.cz"
- DNSRecordType has default value "A"
- BucketName has default value "pycon.cz"
- RedirectHostName has default value "cz.pycon.org"

These values can be changed during the stack creation.

First create a stack to deploy S3 bucket, than a stack to deploy Hosted Zone.

Optionally you can use AWS CLI create-stack command, here is an example:
- create a stack to deploy a S3 bucket.

`aws cloudformation create-stack --stack-name cz-pycon-2015-s3-bucket --template-body file://{{ some_local_path }}/aws_cf_templates/cz-pycon-2015-s3-bucket.template`
- create a stack to deploy a hosted zone

`aws cloudformation create-stack --stack-name cz-pycon-2015-hosted-zone --template-body file://{{ some_local_path }}/aws_cf_templates/cz-pycon-2015-hosted-zone.template`

Full documentation for AWS CLI create-stack command can be found here: http://docs.aws.amazon.com/cli/latest/reference/cloudformation/create-stack.html

Requirements:
There is only one: For the S3 bucket name value please use the exact, full domain, this is important! (e.g. pycon.cz)

**Comment by tricoder42** [comment]
Thank you! :rocket:

**Pull request merged**